[CUSTOMER]  Massive operation simulation

quantranhong1999 opened the pull request.

quantranhong1999 committed:
[CUSTOMER] Massive operation simulation

quantranhong1999 committed:
fixup! [CUSTOMER] Massive operation simulation

Should create sub mailboxes from mailboxDName instead of mailboxDNewName

quantranhong1999 committed:
fixup! fixup! [CUSTOMER] Massive operation simulation

- Variablize numberOfSubMailboxes
- Clean up sub mailboxes

chibenwa reviewed (approved):
Please share a run result for this ;-)

quantranhong1999 committed:
fixup! fixup! fixup! [CUSTOMER] Massive operation simulation

Before: no pause => renameMailbox seems cause an inconsistent problem make deleteMailbox randomly fail cause some mailbox is not found.

After: 5s pause => enough to the maiboxes' name eventually changed => all delete requests succeed.

Arsnael commented on src/main/scala-2.12/org/apache/james/gatling/imap/scenari/MassiveOperationScenario.scala:
```diff
@@ -0,0 +1,78 @@
+    .exec(repeat(numberOfSubMailboxes, "loopId")(pause(gracePeriod).exec(exec(imap("createFolder").createFolder(s"$mailboxDName.$${loopId}").check(ok)))))
+  private val renameMailboxD = exec(imap("renameFolder").renameFolder(mailboxDName, mailboxDNewName).check(ok))
+  private val deleteMailboxD = exec(imap("deleteFolder").deleteFolder(mailboxDNewName).check(ok))
+    .exec(repeat(numberOfSubMailboxes, "loopId")(pause(gracePeriod).exec(exec(imap("deleteFolder").deleteFolder(s"$mailboxDNewName.$${loopId}").check(ok)))))
```
So it seems in IMAP if we delete a folder, it retroactively deletes all subfolders... (or so I think) so why do you need to try delete those subfolders after? those that work?

quantranhong1999 replied:
> those that work?

Deleting subfolders explicitly did work on James.

Arsnael replied:
Ok... not sure to remember how it works then :D

chibenwa replied:
When you don't know read the spec, Luke!

https://datatracker.ietf.org/doc/html/rfc3501#section-6.3.4

```
      The DELETE command MUST NOT remove inferior hierarchical names.
      For example, if a mailbox "foo" has an inferior "foo.bar"
      (assuming "." is the hierarchy delimiter character), removing
      "foo" MUST NOT remove "foo.bar".  It is an error to attempt to
      delete a name that has inferior hierarchical names and also has
      the \Noselect mailbox name attribute (see the description of the
      LIST response for more details).
  ```
